File: 1127-last-stone-weight/last-stone-weight.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& s) {
       priority_queue<int> pq(s.begin(),s.end());
       while(pq.size() > 1){
           int y = pq.top();
           pq.pop();
           int x = pq.top();
           pq.pop();
           if(y > x){
               pq.push(y-x);
           }
       }
       return pq.size()? pq.top() : 0 ;
    }
};

File: 1127-last-stone-weight/last-stone-weight.rs
use std::collections::BinaryHeap;
impl Solution {
    pub fn last_stone_weight(stones: Vec<i32>) -> i32 {
       let mut max_heap:BinaryHeap<i32> =  BinaryHeap::from(stones);
       while max_heap.len() > 1 {
           let y = max_heap.pop().unwrap();
           let x = max_heap.pop().unwrap();
           if y > x {
               let mut z = y - x;
               max_heap.push(z);
           }
       }
       if max_heap.len() == 1 {max_heap.pop().unwrap()} else {0}
    }
}